src/ui/services/ipc/ipcUiService.ts

```diff
@@ -76,21 +82,46 @@ export class IpcUiService {
     workerChannel.start();
 
     // Say hi.
-    this.workerSendMessage({
-      type: IpcWorkerMessageType.UTILS_PING,
-      data: {
-        message: 'Hello from UI',
-      },
-    });
+    const resp: IpcWorkerResponseOf<IpcWorkerMessageType.UTILS_PING> =
```

Comment:
This whole PR is for this line.

The UI can call the worker, async, and gets a typed response.